### DISCUSSION TITLE
Azure Synapse temp table issues

### DISCUSSION THREAD

**Discussion by Shubhkishore** [great-expectations/great_expectations#5078] [Support]
Hi,

I am trying to use Great Expectations with Azure synapse dedicated sql pool. I am running into issues when trying to create an expectation suite. The error is as follows:

```
ProgrammingError: ('42S02', "[42S02] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid object name 'tempdb.pdw.INFORMATION_SCHEMA_COLUMNS'. (208) (SQLExecDirectW)")
```

```
ProgrammingError: (pyodbc.ProgrammingError) ('42S02', "[42S02] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid object name 'tempdb.pdw.INFORMATION_SCHEMA_COLUMNS'. (208) (SQLExecDirectW)")
[SQL: SELECT tempdb.[INFORMATION_SCHEMA].[COLUMNS].[TABLE_NAME] 
FROM tempdb.[INFORMATION_SCHEMA].[COLUMNS] 
WHERE tempdb.[INFORMATION_SCHEMA].[COLUMNS].[TABLE_NAME] LIKE CAST(? AS NVARCHAR(max))]
[parameters: ('#ge_temp_5f5d7fe6[_][_][_]%',)]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

I see that Azure synapse [has been used successfully](https://greatexpectations.io/case-studies/avanade-case-study/) with Great Expectations. Could someone please help me with this? 🙇 

EDIT: It seems like the expectation suite gets created if the expectations are at table level but I'm getting the above error when there are column level expectations present. Will Appreciate any help on this!

**Comment by austiezr** [comment]
Hey @Shubhkishore ! Thanks for reaching out; we'll review internally and be in touch.

**Reply by austiezr** [reply]
You may try passing `create_temp_table` as `False` in a `batch_spec_passthrough` in your batch request:

```
...
    'batch_spec_passthrough': {
        'create_temp_table': False
    }
...
```

**Reply by Shubhkishore** [reply]
Thanks @austiezr 
Seems like the solution you suggested is for V3 API. I was wondering if there is a similar alternative present for V2 API. More details are present in [this issue](https://github.com/great-expectations/great_expectations/issues/5079)
